### PR TITLE
Fix for IDENTITY-4372

### PR DESCRIPTION
**Pull request by Prakhash**

### CODE REVIEW

File: components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 import org.apache.commons.lang.StringUtils;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.context.CarbonContext;
```
Comment (cdwijayarathna) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java]: Shouldn't we use PriviledgedCarbonContext? please check

Comment (Prakhash) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java]: We can use CarbonContext for get. I have formatted the code`

Comment (darshanasbg) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java]: This the first place you introduce CarbonContext in the listener code. Other places its picks the tenant from the user store manager instance. Better to follow the same approach here as well. Ref following,

`String tenantDomain = IdentityTenantUtil.getTenantDomain(userStoreManager.getTenantDomain());`

Comment (Prakhash) [inline components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java]: Made the changes using the suggested code

**Comment by thariyarox** [comment]
Reviewed the fix. Please merge.

**Pull request merged**